### PR TITLE
PAPP-19195: SaaS compatibility

### PR DESCRIPTION
**Pull request by pzhou-splunk**
* Compatibility changes for Python 3 support [PAPP-19196]
* Add new optional asset configuration parameters 'ers\_user' and 'ers\_password'
* Add new optional asset configuration parameter 'ha\_device' for secondary node [PAPP-21216]
* Updated logo for the app
* Updated documentation for the app
* Added support for the below mentioned actions:
    + list resources [PAPP-8694]
    + get resources [PAPP-8693]
    + create resources [PAPP-8697]
    + update resources [PAPP-8695]
    + delete resources [PAPP-8696]
    + apply policy
    + clear policy
    + list endpoints
    + get device info
    + update device info

### NOTE: Static job is failing because of FIPS compliance flag.

### CODE REVIEW

File: __init__.py [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # File: __init__.py
 #
-# Copyright (c) 2014-2018 Splunk Inc.
+# Copyright (c) 2014-2021 Splunk Inc.
```
Comment (ngaglani-crest) [inline __init__.py:3]: Copyright years are mismatching.

File: ciscoise.json [hunk 2]
```diff
@@ -1,33 +1,35 @@
...
     "product_vendor": "Cisco Systems",
     "product_name": "Cisco ISE",
-    "product_version_regex": "2.2",
-    "min_phantom_version": "3.0.190",
+    "product_version_regex": "/([2].[67])|([3].[01])/",
+    "min_phantom_version": "5.0.0",
     "logo": "logo_cisco.svg",
     "logo_dark": "logo_cisco_dark.svg",
-    "license": "Copyright (c) Phantom Cyber Corporation, 2014-2018",
+    "license": "Copyright (c) 2016-2021 Splunk Inc.",
```
Comment (ngaglani-crest) [inline ciscoise.json:17]: Copyright years are mismatching. It should be either `2016-2021` or `2014-2021` across all the files.

File: ciscoise.json [hunk 3]
```diff
@@ -113,14 +119,13 @@
...
-                        "ise endpoint id"
+                        "ise endpoint id",
+                        "ise resource id"
                     ]
                 },
                 {
                     "data_path": "action_result.data.*.SearchResult.resources.*.link.href",
                     "data_type": "string",
-                    "column_name": "Link to ISE Endpoint",
-                    "column_order": 2,
```
Comment (ngaglani-crest) [inline ciscoise.json:123]: Is there any specific reason for removing this from the output table?

File: ciscoise.json [hunk 4]
```diff
@@ -801,13 +925,866 @@
...
+            "versions": "EQ(*)"
+        },
+        {
+            "action": "get resources",
+            "description": "Search for resources. Returns resource with `resource id`, else finds resources with key-value",
+            "type": "investigate",
+            "identifier": "get_resources",
+            "read_only": true,
+            "parameters": {
+                "resource": {
```
Comment (ngaglani-crest) [inline ciscoise.json:1080]: Please update the parameters' description.

File: ciscoise.json [hunk 5]
```diff
@@ -801,13 +925,866 @@
...
+            },
+            "output": [
+                {
+                    "data_path": "action_result.parameter.ip_mac_address",
+                    "data_type": "string",
+                    "contains": [
+                        "mac address",
+                        "ip"
+                    ],
+                    "column_name": "ip mac address",
```
Comment (ngaglani-crest) [inline ciscoise.json:1669]:
```suggestion
                    "column_name": "IP MAC Address",
```

File: ciscoise.json [hunk 6]
```diff
@@ -801,13 +925,866 @@
...
+                    "column_order": 0
+                },
+                {
+                    "data_path": "action_result.parameter.policy_name",
+                    "data_type": "string",
+                    "column_name": "policy name",
```
Comment (ngaglani-crest) [inline ciscoise.json:1675]:
```SUGGESTION
                    "column_name": "Policy Name",
```

File: ciscoise.json [hunk 7]
```diff
@@ -801,13 +925,866 @@
...
+                    "column_order": 1
+                },
+                {
+                    "data_path": "action_result.message",
+                    "data_type": "string",
+                    "example_values": [
+                        "Policy applied"
+                    ],
+                    "column_name": "status",
```
Comment (ngaglani-crest) [inline ciscoise.json:1684]: Same as above.

File: ciscoise.json (hunk 3 above)
Comment (pdesai-crest) [reply ciscoise.json:123]: Removed this because the link in the column was broken. Was leading to nowhere

File: ciscoise.json (hunk 2 above)
Comment (pdesai-crest) [reply ciscoise.json:17]: Updated

File: __init__.py (hunk 1 above)
Comment (pdesai-crest) [reply __init__.py:3]: updated

File: ciscoise.json (hunk 4 above)
Comment (pdesai-crest) [reply ciscoise.json:1080]: updated

File: ciscoise.json (hunk 6 above)
Comment (pdesai-crest) [reply ciscoise.json:1675]: updated

File: ciscoise.json (hunk 7 above)
Comment (pdesai-crest) [reply ciscoise.json:1684]: updated

File: ciscoise.json (hunk 5 above)
Comment (pdesai-crest) [reply ciscoise.json:1669]: updated

File: ciscoise.json [hunk 8]
```diff
@@ -801,13 +925,866 @@
...
+                    "example_values": [
+                        1
+                    ]
+                }
+            ],
+            "versions": "EQ(*)"
+        },
+        {
+            "action": "get resources",
+            "description": "Search for resources. Returns resource with `resource id`, else finds resources with key-value",
```
Comment (dgopani-crest) [inline ciscoise.json:1075]: Can we change it with "Get the information about resource if resource_id is provided. Fetch the list of resources match with the key-value filter"

File: ciscoise_consts.py [hunk 9]
```diff
@@ -54,10 +53,24 @@
 CISCOISE_SUCC_LIST_ENDPOINTS = "{0} Endpoints found"
 CISCOISE_SUCC_GET_ENDPOINT = "Endpoint found"
 CISCOISE_SUCC_UPDATE_ENDPOINT = "Endpoint updated"
+CISCOISE_ERR_INVALID_PARAM = "Please provide non-zero positive integer in {param}"
```
Comment (dgopani-crest) [inline ciscoise_consts.py:56]: Please provide a non-zero positive integer in {param}

File: ciscoise_connector.py [hunk 10]
```diff
@@ -48,83 +55,139 @@ def __init__(self):
...
+                self.debug_print("Making call to secondary device")
+                ret_val, ret_data = func(try_ha_device=True, *args, **kwargs)
+
+            return ret_val, ret_data
+
+        return make_another_call
+
+    def _call_ers_api(self, endpoint, action_result, data=None, allow_unknown=True, method="get", try_ha_device=False):
+        if self._ers_auth is None:
+            return action_result.set_status(phantom.APP_ERROR, "Please provide ERS credentials for this action"), None
```
Comment (dgopani-crest) [inline ciscoise_connector.py:95]: Can we move messages to constant and also update it with "ERS credentials in asset configuration are required for this action"

File: exclude_files.txt [hunk 11]
```diff
@@ -1,6 +1 @@
-docker-compose.yml
-Makefile
 .git*
```
Comment (dgopani-crest) [inline exclude_files.txt:1]: delete this file. app is opensource now

**Comment by dgopani-crest** [comment]
Spelling mistake in JSON 2 places "informaton" -> "information"

File: ciscoise.json (hunk 8 above)
Comment (pdesai-crest) [reply ciscoise.json:1075]: Updated

File: ciscoise_consts.py (hunk 9 above)
Comment (pdesai-crest) [reply ciscoise_consts.py:56]: Updated

File: ciscoise_connector.py (hunk 10 above)
Comment (pdesai-crest) [reply ciscoise_connector.py:95]: Updated

File: exclude_files.txt (hunk 11 above)
Comment (pdesai-crest) [reply exclude_files.txt:1]: deleted

File: ciscoise.json (hunk 2 above)
Comment (dgopani-crest) [reply ciscoise.json]: @pdesai-crest, It looks like release_notes has the first entry for 2016 and we have updated with 2014-2021 in all the files. Can you check with the US team?

**Review by dgopani-crest — APPROVED** [review]

**Review by joshquerinphantom — APPROVED** [review]

**Review by brian-phantom — APPROVED** [review]

**Pull request merged**